### PR TITLE
feat: Create Theme auto-set feature

### PR DESCRIPTION
**Pull request by gabrielfs7**
Ticket: https://oat-sa.atlassian.net/browse/TR-2966

## Description

- Create theme auto-set feature

## TODO 

- [x] Unit tests
- [ ] Update composer with packages

## PRs

- https://github.com/oat-sa/extension-tao-testqti/pull/2218
- https://github.com/oat-sa/extension-tao-bosa-selor/pull/118
- https://github.com/oat-sa/extension-tao-delivery-rdf/pull/420
- https://github.com/oat-sa/tao-core/pull/3444

### CODE REVIEW

File: model/theme/Listener/ThemeAutoSetListener.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+namespace oat\taoDeliveryRdf\model\theme\Listener;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoDeliveryRdf\model\event\DeliveryCreatedEvent;
+use oat\taoDeliveryRdf\model\theme\Exception\ThemeAutoSetNotSupported;
+use oat\taoDeliveryRdf\model\theme\Service\ThemeAutoSetService;
+use Throwable;
+
+class ThemeAutoSetListener extends ConfigurableService
```
Comment (gabrielfs7) [inline model/theme/Listener/ThemeAutoSetListener.php:31]: Cannot migrate to DI container without big refactoring :/

File: model/theme/Listener/ThemeAutoSetListener.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+{
+    public const SERVICE_ID = 'taoDeliveryRdf/ThemeAutoSetListener';
```
Comment (gabrielfs7) [inline model/theme/Listener/ThemeAutoSetListener.php:33]: I am forced to use a SERVICE_ID here

**Review by hectoras — APPROVED** [review]

File: model/theme/Service/ThemeAutoSetService.php [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    private $qtiTestRepository;
+
+    /** @var ThemeServiceAbstract */
+    private $themeService;
+
+    public function __construct(
+        Ontology                     $ontology,
+        DeliveryThemeDetailsProvider $deliveryThemeDetailsProvider,
+        QtiTestRepositoryInterface   $qtiTestRepository,
+        ThemeServiceAbstract         $themeService
```
Comment (shpran) [inline model/theme/Service/ThemeAutoSetService.php:52]:
```suggestion
        Ontology $ontology,
        DeliveryThemeDetailsProvider $deliveryThemeDetailsProvider,
        QtiTestRepositoryInterface $qtiTestRepository,
        ThemeServiceAbstract $themeService
```

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**